images/gh-gl-sync/SpackCIBridge.py

```diff
@@ -164,59 +164,70 @@ def list_github_prs(self):
+                            print(f"Skip pushing {pr_string} because api says HEAD is {a_sha}, "
+                                  f"while repo says HEAD is {r_sha}")
+                            backlogged = f"GitHub HEAD shas out of sync (repo={r_sha}, API={a_sha})"
+                            push = False
+                        # Check if our PR's merge base is an ancestor of the latest tested main branch commit.
+                        elif subprocess.run(
+                                ["git", "merge-base", "--is-ancestor", merge_base_sha, self.latest_tested_main_commit]
+                                ).returncode == 0:
+                            print(f"{tmp_pr_branch}'s merge base IS an ancestor of latest_tested_main "
+                                  f"{merge_base_sha} vs. {self.latest_tested_main_commit}")
+                            try:
+                                subprocess.run(["git", "checkout", self.latest_tested_main_commit], check=True)
+                                subprocess.run(["git", "checkout", "-b", pr_string], check=True)
+                                commit_msg = f"Merge {pull.head.sha} into {self.latest_tested_main_commit}"
+                                subprocess.run(
+                                    ["git", "merge", "--no-ff", "-m", commit_msg, tmp_pr_branch],
+                                    check=True)
+                                print(f"Merge succeeded, ready to push {pr_string} to GitLab for CI pipeline testing")
+                            except subprocess.CalledProcessError:
+                                print("Failed to merge PR {0} ({1}) with latest tested {2} ({3}). Skipping"
+                                      .format(pull.number, pull.head.ref, self.main_branch, self.latest_tested_main_commit))
+                                self.unmergeable_shas.append(pull.head.sha)
+                                subprocess.run(["git", "merge", "--abort"])
+                                backlogged = "merge conflicts with {}".format(self.main_branch)
+                                push = False
+                                continue
+                        else:
+                            print(f"Skip pushing {pr_string} because its merge base is NOT an ancestor of "
+                                  f"latest_tested_main {merge_base_sha} vs. {self.latest_tested_main_commit}")
+                            backlogged = "base"
+                            push = False
                     else:
-                        print(f"Skip pushing {pr_string} because its merge base is NOT an ancestor of "
-                              f"latest_tested_main {merge_base_sha} vs. {self.latest_tested_main_commit}")
-                        backlogged = "base"
-                        push = False
+                        try:
+                            subprocess.run(["git", "fetch", "--unshallow", "github",
```

Comment:
Looks like you need to push another commit anyway to fix some failing CI.  Do you mind just adding a comment here that this is the old case, where we just push whatever merge commit github makes for us?